hamilton/lifecycle/conditional_tqdm.py

```diff
@@ -0,0 +1,105 @@
+from typing import Any, Dict, List, Optional
+
+import tqdm
+
+from hamilton import graph, node
+from hamilton.lifecycle import base
+
+
+class TQDMHook(
```

Comment:
Maybe call it `ProgressBarHook`?

Reply:
TQDM is an implementation detail... OTOH we could have other progress bars.

Reply:
Nah I think it's specific because it requires a dependency.